fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -311,11 +339,30 @@ def sql(self):
         be produced.  This next line converts the list of rows of the table in the diagram to a list of columns. Each 
         set of queries in a column are then reduced to a single data blending sql query. 
         """
-        querysets_tx = list(
-            zip(*[dataset_query.sql for dataset_query in dataset_queries])
-        )
 
-        return [
-            _blend_query(self._dimensions, metrics, self.orders, field_maps, queryset)
-            for queryset in querysets_tx
-        ]
+        # TODO: what if some of the datasets end up doing more reference/total calculations? Make sure to prevent this.
+        per_dataset_queries_count = max(
+            [len(dataset_queries) for dataset_queries in datasets_queries]
+        )
+        query_sets = [[] for _ in range(per_dataset_queries_count)]
+
+        for dataset_queries in datasets_queries:
+            for i, dataset_query in enumerate(dataset_queries):
+                query_sets[i].append(dataset_query)
```

Comment:
Just want to check - but I am guessing the order of the queries lines up with the above large comment? e.g. [base query, reference query, totals query, totals and reference query]?

Reply:
The transformation that is described in that large comment still applies yes. But it's possible there are more queries than just those 4.

Reply:
It's possible that I already solved my own TODO that I added here, but I'm not entirely sure.